In [3]:
import pandas as pd

In [4]:
data = pd.read_csv('train.csv')

In [10]:
print(f"Dimensions: {data.shape[0]} rows, {data.shape[1]} columns\n")
data.info()

Dimensions: 891 rows, 12 columns

<class 'pandas.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    str    
 4   Sex          891 non-null    str    
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    str    
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    str    
 11  Embarked     889 non-null    str    
dtypes: float64(2), int64(5), str(5)
memory usage: 83.7 KB


In [12]:
print("\n--- Summary Statistics ---")
print(data.describe())


--- Summary Statistics ---
       PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200  


### Initial Data Inspection Summary

The Titanic training dataset contains 891 passenger records across 12 distinct columns, splitting features into explicit numerical distributions and descriptive categorical strings. Initial profiling via `.info()` highlights critical structural gaps: both `Age` and `Cabin` suffer from significant missing data, with `Cabin` missing the vast majority of its entries, requiring aggressive preprocessing or feature engineering later. Synthesizing the statistical summary via `.describe()`, we see that while survival is explicitly structured as a binary integer, the average survival rate sits at roughly 38.4%. Furthermore, the `Fare` attribute exhibits extreme right-skewness, ranging from free passage ($0.0) up to a maximum outlier of over $512.3, which suggests potential layout disparities across passenger classes.